massa-bootstrap/src/lib.rs

```diff
@@ -349,12 +392,19 @@ impl BootstrapServer {
                     // load cache if absent
                     if bootstrap_data.is_none() {
                         massa_trace!("bootstrap.lib.run.select.accept.cache_load.start", {});
+
+                        // Note that all requests are done simultaneously except for the consensus graph that is done after the others.
+                        // This is done to ensure that the execution bootstrap state is older than the consensus state.
+                        // If the consensus state snapshot is older than the execution state snapshot,
+                        //   the execution final ledger will be in the future after bootstrap, which causes an inconsistency.
                         let get_peers = self.network_command_sender.get_bootstrap_peers();
                         let get_pos_graph = self.consensus_command_sender.get_bootstrap_state();
-                        let (res_peers, res_pos_graph) = tokio::join!(get_peers, get_pos_graph);
+                        let execution_state = self.execution_command_sender.get_bootstrap_state();
+                        let (res_peers, res_execution) = tokio::join!(get_peers, execution_state);
                         let peer_boot = res_peers?;
-                        let (pos_boot, graph_boot) = res_pos_graph?;
-                        bootstrap_data = Some((pos_boot, graph_boot, peer_boot));
+                        let execution_state = res_execution?;
+                        let (pos_boot, graph_boot) = get_pos_graph.await?;
+                        bootstrap_data = Some((pos_boot, graph_boot, peer_boot, execution_state));
```

Comment:
Why do you wrap that in a some, to unwrap it just after ?

Reply:
* I try to get it from cache, if it fails, the value remains None
* if the value is None I load it
* at this point I know it's present and I unwrap it

That's just there to avoid nesting too many "if" expressions